### PR TITLE
Add OtherDetail attribute to MD 502SU XML to include ab code

### PR DESCRIPTION
**Pull request by anisharamnani**
## Link to pivotal/JIRA issue
[link](https://codeforamerica.atlassian.net/browse/FYST-1483)

## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

## What was done?
Add OtherDetail attribute to MD 502SU XML to include ab code
## How to test?
- select Todd 1099INT
- their XML should include: 
* `Form502SU Subtractions OtherDetail Code` with code "ab"
* `Form502SU Subtractions OtherDetail Amount` with the sum of the `interest_on_government_bonds`

Screenshot 2024-12-19 at 5 30 28 PM

### CODE REVIEW

**Review by arinchoi03 — APPROVED** [review]
one comment to remove comments but lgtm!

File: app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb [hunk 1]
```diff
@@ -11,7 +11,13 @@ class Md502Su < SubmissionBuilder::Document
...
               build_xml_doc("Form502SU", documentId: "Form502SU") do |xml|
                 xml.Subtractions do |subtractions|
-                  subtractions.Total calculated_fields.fetch(:MD502_SU_LINE_1)
+                  if calculated_fields.fetch(:MD502_SU_LINE_AB).positive?
+                    subtractions.OtherDetail do |other_detail|
+                      other_detail.Code "ab"
+                      other_detail.Amount calculated_fields.fetch(:MD502_SU_LINE_AB)
+                    end
+                  end
+                  subtractions.Total calculated_fields.fetch(:MD502_SU_LINE_1) # check if this should be here
```
[the remark this answers is not in the document]
Comment (anisharamnani) [reply app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb:20]: ah, @arinchoi03 good catch. these were just my notes that i forgot to remove.

File: app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb [hunk 2]
```diff
@@ -11,6 +11,12 @@ class Md502Su < SubmissionBuilder::Document
             def document
               build_xml_doc("Form502SU", documentId: "Form502SU") do |xml|
                 xml.Subtractions do |subtractions|
+                  if calculated_fields.fetch(:MD502_SU_LINE_AB).positive?
+                    subtractions.OtherDetail do |other_detail|
+                      other_detail.Code "ab"
```
Comment (mpidcock) [inline app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb:16]: Needs to be captial "AB" - it's a enum

**Review by mpidcock — CHANGES_REQUESTED** [review]

File: app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb (hunk 2 above)
Comment (mpidcock) [reply app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb:16]: There is probably a validation error, we should be catching those in tests :(

Comment (anisharamnani) [reply app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb]: Fixed!

Comment (anisharamnani) [reply app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb]: I will add both!

**Review by mpidcock — APPROVED** [review]
speedy gonzales

File: app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb (hunk 2 above)
Comment (arinchoi03) [reply app/lib/submission_builder/ty2024/states/md/documents/md502_su.rb]: oof sorry I missed this in code review

**Pull request merged**